expr/function/into.go

```diff
@@ -0,0 +1,104 @@
+package function
+
+import (
+	"sort"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/field"
+	"github.com/brimdata/zed/zcode"
+)
+
+// https://github.com/brimdata/zed/blob/main/docs/language/functions.md#into
+type Into struct {
+	zcode.Builder
+	keyType zed.Type
+	mapper  *zed.Mapper
+	zctx    *zed.Context
+}
+
+func NewInto(zctx *zed.Context) *Into {
+	return &Into{
+		mapper: zed.NewMapper(zctx),
+		zctx:   zctx,
+	}
+}
+
+func (n *Into) Call(ctx zed.Allocator, args []zed.Value) *zed.Value {
+	val := args[0]
+	typ := zed.TypeRecordOf(val.Type)
+	if typ == nil {
+		return &val
+	}
+	valType := n.mapper.Lookup(typ.ID())
+	if valType == nil {
+		types := collectTypes(typ.Columns)
+		types = dedupeTypes(types)
+		if len(types) == 1 {
+			valType = types[0]
+		} else {
+			valType = n.zctx.LookupTypeUnion(types)
+		}
+		n.mapper.EnterType(typ.ID(), valType)
+	}
+	if n.keyType == nil {
+		n.keyType = n.zctx.LookupTypeArray(zed.TypeString)
+	}
+	n.Reset()
+	n.encode(typ.Columns, valType, field.Path{}, val.Bytes)
+	return ctx.NewValue(n.zctx.LookupTypeMap(n.keyType, valType), n.Bytes()).Copy()
```

Comment:
No need for `Copy` here since the `zcode.Builder.Bytes` caller own the returned slice.
```suggestion
	return ctx.NewValue(n.zctx.LookupTypeMap(n.keyType, valType), n.Bytes())
```